fast64_internal/mk64/__init__.py

```diff
@@ -0,0 +1,152 @@
+import bpy
+from bpy.types import PropertyGroup
+from ..f3d.f3d_material import createF3DMat
+from ..f3d.f3d_gbi import get_F3D_GBI
+from ..f3d.f3d_parser import getImportData, importMeshC
+from ..panels import MK64_Panel
+from ..utility import prop_split
+from bpy.utils import register_class, unregister_class
+from ..utility import raisePluginError
+from .f3d_course_parser import MK64F3DContext, parseCourseVtx
+
+
+class MK64_Properties(PropertyGroup):
+    """Global MK64 Scene Properties found under scene.fast64.mk64"""
+
+    # Import Course DL
+    EnableRenderModeDefault: bpy.props.BoolProperty(name="Enable Render Mode by Default", default=True)
+
+    @staticmethod
+    def upgrade_changed_props():
+        pass
+
+
+class MK64_ImportCourseDL(bpy.types.Operator):
+    # set bl_ properties
+    bl_idname = "object.fast64_mk64_course_import_dl"
+    bl_label = "Import Course DL"
+    bl_options = {"REGISTER", "UNDO", "PRESET"}
+
+    # Called on demand (i.e. button press, menu item)
+    # Can also be called from operator search menu (Spacebar)
+    def execute(self, context):
+        obj = None
+        if context.mode != "OBJECT":
+            bpy.ops.object.mode_set(mode="OBJECT")
+
+        try:
+            name = context.scene.DLImportName
+            importPath = bpy.path.abspath(context.scene.DLImportPath)
+            basePath = bpy.path.abspath(context.scene.DLImportBasePath)
+            scaleValue = context.scene.blenderF3DScale
+
+            removeDoubles = context.scene.DLRemoveDoubles
+            importNormals = context.scene.DLImportNormals
+            drawLayer = context.scene.DLImportDrawLayer
+
+            paths = [importPath]
+
+            if "course_data" in importPath:
+                paths += [importPath.replace("course_data", "course_displaylists.inc")]
+
+            paths += [
+                importPath.replace("course_data", "course_textures.linkonly").replace(
+                    "course_displaylists.inc", "course_textures.linkonly"
+                )
+            ]
+
+            data = getImportData(paths)
+
+            material = createF3DMat(None)
+
+            material.f3d_mat.rdp_settings.set_rendermode = context.scene.fast64.mk64.EnableRenderModeDefault
```

Comment:
Obviously I might be wrong here but should you not be setting the default combiner
```suggestion
            f3d_mat = material.f3d_mat
            f3d_mat.rdp_settings.set_rendermode = context.scene.fast64.mk64.EnableRenderModeDefault
            f3d_mat.combiner1.A = 'TEXEL0'
            f3d_mat.combiner1.B = '0'
            f3d_mat.combiner1.C = 'SHADE'
            f3d_mat.combiner1.D = '0'
            f3d_mat.combiner1.A_alpha = 'TEXEL0'
            f3d_mat.combiner1.B_alpha = '0'
            f3d_mat.combiner1.C_alpha = 'SHADE'
            f3d_mat.combiner1.D_alpha = '0'
            f3d_mat.combiner2.name = ''
            f3d_mat.combiner2.A = 'TEXEL0'
            f3d_mat.combiner2.B = '0'
            f3d_mat.combiner2.C = 'SHADE'
            f3d_mat.combiner2.D = '0'
            f3d_mat.combiner2.A_alpha = 'TEXEL0'
            f3d_mat.combiner2.B_alpha = '0'
            f3d_mat.combiner2.C_alpha = 'SHADE'
            f3d_mat.combiner2.D_alpha = '0'
```

Reply:
hmm without setting this it work so I think the material create by createF3DMat set the default value of f3d_mat, so I don't see what this fix ?

Reply:
hmm work fine without it and it's change in displaylist